synapse/storage/databases/main/user_directory.py

```diff
@@ -441,7 +441,9 @@ async def is_room_world_readable_or_publicly_joinable(self, room_id: str) -> boo
             (EventTypes.RoomHistoryVisibility, ""),
         )
 
-        current_state_ids = await self.get_filtered_current_state_ids(  # type: ignore[attr-defined]
+        # Getting the partial state is fine, as we're not looking at membership
+        # events.
```

Comment:
Soooo, the reason I didn't do that is because it means that the stores then need to call back up to the `StateStorage`, which ends up with circular dependency and in general is a bit annoying. We can't easily lift the function out of the data stores as its a background update. :shrug: